src/observer.ts

```diff
@@ -64,20 +64,37 @@ function getArnsResolution({
   });
   const dataHash = crypto.createHash('sha256');
 
+  let streamBytesProcessed = 0;
+  const MAX_BYTES_TO_PROCESS = 1048576; // 1MiB
```

Comment:
nit: i'd say we could make this configurable but someone might try and set it to something ridiculously small so fine as a const

Reply:
I don't think it should be configurable as all observers should hash the same data